_delphi_utils_python/delphi_utils/validator/validate.py

```diff
@@ -35,7 +35,8 @@ def __init__(self, params):
         self.suppressed_errors = [ValidationFailure(**entry) for entry in suppressed_errors]
 
         # Date/time settings
-        self.time_window = TimeWindow.from_params(validation_params["common"]["end_date"],
+        end_date = end_date_helper(validation_params)
```

Comment:
Perhaps simpler to just directly save the calculated `end_date` to validation params. Saving to `validation_params` lets us pass the calculated `end_date` directly to the dynamic and static validation objects, rather than re-calculating `end_date` there.

```suggestion
        validation_params["end_date"] = end_date_helper(validation_params)
```